stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1093,10 +1167,13 @@ internal constructor(
             userCredentialStorage.clear()
         }
         lifecycleObserver.dispose()
-        appSettingsManager.clear()
         _repositoryFacade = null
-        val currentJob = currentCoroutineContext()[Job]
-        scope.coroutineContext.cancelChildrenExcept(currentJob)
+        appSettingsManager.clear()
+
+        clientState.toMutableState()?.clearState()
+        clientState.toMutableState()?.setInitializionState(InitializationState.NOT_INITIALIZED)
+
+        logger.v { "[disconnectSuspend] completed('$userId')" }
```

Comment:
Why did we rearranged this method? :thinking:

Reply:
To set `NOT_INITIALIZED` only when the disconnection is fully complete.